src/components/worker/worker.py

```diff
@@ -109,65 +109,203 @@ def __init__(
+                if color_node:
+                    color_node.isp.link(sync.inputs["rgb"])
+                pointcloud.outputPointCloud.link(sync.inputs["pcl"])
+                sync.out.link(xOut.input)
+
+        self.pipeline = dai.Pipeline()
+        try:
+            stage = "color"
+            color_node = configure_color()
+
+            stage = "stereo"
+            depth_node = configure_stereo()
+
+            stage = "point cloud"
+            configure_pc(color_node, depth_node)
+
+            LOGGER.info("Successfully configured OakCamera")
+        except Exception as e:
+            msg = f"Error configuring OakCamera at stage '{stage}': {e}"
+            resolution_err_substr = "bigger than maximum at current sensor resolution"
+            calibration_err_substr = "no Camera data available"
+            if resolution_err_substr in str(e):
+                msg += ". Please adjust 'height_px' and 'width_px' in your config to an accepted resolution."
+            elif calibration_err_substr in str(e):
+                msg += ". If using a non-integrated model, please check that the camera is calibrated properly."
+            LOGGER.error(msg)
 
         self.message_synchronizer = MessageSynchronizer()
         self.configured = True
 
-    def start(self):
-        if self.oak:
-            self.oak.start()
-            LOGGER.info("Started OakCamera")
-        else:
-            raise AttributeError(
-                "oak.start() called before oak was assigned. Must configure worker first."
+    async def start(self):
+        self.device = None
+        while not self.device and self.should_exec:
+            try:
+                self.device = dai.Device(self.pipeline)
+                LOGGER.info("Successfully initialized OakCamera.")
+            except Exception as e:
+                LOGGER.error(f"Error initializing OakCamera: {e}")
+                await asyncio.sleep(1)
+
+        self.color_sensor_queues: List[SensorAndQueue] = []
+        for cs in self.cfg.sensors.color_sensors:
+            q = self.device.getOutputQueue(cs.get_unique_name(), 30, blocking=False)
+            self.color_sensor_queues.append(SensorAndQueue(cs, q))
+
+        if self.cfg.sensors.stereo_pair:
+            self.depth_queue = self.device.getOutputQueue(
+                self.depth_stream_name, 30, blocking=False
+            )
+            if self.user_wants_pc:
+                self.pcd_queue = self.device.getOutputQueue(
+                    self.pc_stream_name, 5, blocking=False
+                )
+
+        should_get_synced_color_depth_outputs = len(
+            self.color_sensor_queues
+        ) == 1 and bool(self.depth_queue)
+        if should_get_synced_color_depth_outputs:
+            self.color_sensor_queues[0].queue.addCallback(
+                self.message_synchronizer.add_color_msg
             )
+            self.depth_queue.addCallback(self.message_synchronizer.add_depth_msg)
```

Comment:
It's honestly great that they created this callback logic in the new SDK! It should give us a small performance bump since we aren't grabbing the lock and copying out the entire queue every time we ask for new messages.